spec.yaml

```diff
@@ -393,72 +393,134 @@ definitions:
+                    MAY include a secret hash in the path, in which case there MAY be
+                    no associated `sharedSecret`.
             webapp:
               type: object
               properties:
                 uriTemplate:
                   type: string
+                  description: |
+                    A templated URI to a client-browsable view of the shared resource,
+                    such that users may use the web applications available at the site.
+                    The URI MAY include a secret hash in the path. If the path includes
+                    a `{relative-path-to-shared-resource}` placeholder, implementations
+                    MAY replace it with the actual path to ease user interaction.
                 viewMode:
                   type: string
+                  description: |
+                    The permissions granted to the sharee.
+                    - `view` allows access to the web app in view-only mode.
+                    - `read` allows read and download access via the web app.
+                    - `write` allows full editing rights via the web app.
                   enum: ["view", "read", "write"]
+                sharedSecret:
+                  type: string
+                  description: |
+                    An optional secret to be used to access the remote web app,
+                    for example in the form of a bearer token.
             datatx:
               type: object
               properties:
                 sharedSecret:
                   type: string
+                  description: |
+                    An optional secret to be used to access the resource,
+                    for example in the form of a bearer token.
                 srcUri:
                   type: string
+                  description: |
+                    An URI to access the remote resource. The URI MAY be relative,
+                    in which case the prefix exposed by the `/ocm-provider` endpoint MUST
+                    be used, or it may be absolute (recommended). Additionally, the
+                    URI MAY include a secret hash in the path.
                 size:
                   type: integer
         example:
-          webdav:
-            sharedSecret: secret
-            permissions: ["read"]
-            uri: "https://open-cloud-mesh.org/remote.php/webdav/path/to/resource"
-          webapp:
-            uriTemplate: "https://open-cloud-mesh.org/s/share-hash/{relative-path-to-shared-resource}"
-            viewMode: "write"
-          datatx:
-            sharedSecret: secret
-            srcUri: "https://open-cloud-mesh.org/remote.php/webdav/path/to/resource"
-            size: 1000000000000
+          singleProtocolLegacy:
+            name: "webdav"
+            options:
+              sharedSecret: "hfiuhworzwnur98d3wjiwhr"
+              permissions: "some permissions scheme"
```

Comment:
@smesterheide any news about the permissions scheme used in Nextcloud? If not, I propose to leave this in the example as a generic placeholder, it would fit the current as well as future implementations. In this case, can you please approve the PR?

Reply:
As discussed live, we keep it like that